src/App.tsx

```diff
@@ -8,7 +7,52 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { Filter } from './enums/Filter';
+
+import { getTodos } from './api';
+import { debounce } from './helpers/debounce';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [userId, setUserId] = useState(0);
+  const [filterValue, setFilterValue] = useState(Filter.All);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 500), []);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      await getTodos()
+        .then(todoList => todoList
+          .filter(todoItem => {
+            switch (filterValue) {
+              case Filter.Active:
+                return !todoItem.completed;
+
+              case Filter.Completed:
+                return todoItem.completed;
+
+              default:
+                return todoItem;
+            }
+          }))
+        .then(result => setTodos(result));
+    };
+
+    fetchData();
+  }, [filterValue]);
```

Comment:
why do you need to load todos from server each time the `filterValue` is changed? please load them once from API and just use `useMemo` hook to filter by filter change